src/wp-includes/class-wp-admin-bar.php

```diff
@@ -18,23 +18,6 @@ class WP_Admin_Bar {
 	private $bound = false;
 	public $user;
 
-	/**
-	 * @since 3.3.0
-	 *
-	 * @param string $name
-	 * @return string|array|void
-	 */
-	public function __get( $name ) {
-		switch ( $name ) {
```

Comment:
I've researched if these properties are being used (including WP plugins), and it looks like they can be safely removed without breaking BC.

Reply:
@anton-vlasenko is there a search query you used to investigate plugins and themes? If no, how did you investigate usage?

Reply:
That's a good question, @hellofromtonya.
Unfortunately, I'm not writing down every search query that I use (if we are talking about wpdirectory).
I committed these changes sometime last year, so I can't recall.
The best I can do is to search again and post the queries here.

Reply:
@hellofromtonya 
I've changed my mind, and I brought `WP_Admin_Bar::$menu` back.
There is no reliable way to check all the plugins and themes because there are too many search results.
https://wpdirectory.net/search/01GSTW1X69TBN8FH3SY7V8KPY5 returns only a few results, all of which seem good (meaning that `WP_Admin_Bar::$menu` can be removed).
But if I use slightly broader search criteria (https://wpdirectory.net/search/01GSTW68SM7MNRZD0Z9AVBH6R0) I get about 4800 results. It's unrealistic to check all or at least the most popular plugins/themes.
So, to stay safe, I'd prefer to keep the `WP_Admin_Bar::$menu` property for now.

Reply:
Thank you for the search queries @anton-vlasenko. 

Reviewing the first query for `menu` shows that at least 1 plugin (s2 Member Framework) is using this property.

As [discussed in the livestream](https://www.youtube.com/live/vDZWepDQQVE?feature=share&t=9362), the plan is:

* Remove the `__get()`.
* Since `menu` is being used, add it as a public property on the class with a default value of `array()`.
```php
	/**
	 * Deprecated menu property.
	 *
	 * @since 3.1.0
	 * @deprecated 3.3.0 Modify admin bar nodes with WP_Admin_Bar::get_node(), WP_Admin_Bar::add_node(), and WP_Admin_Bar::remove_node().
	 * @var array
	 */
	public $menu = array();
```
* What about `proto` (for protocol) property? 

From the livestream, there was consensus to remove it if _unused_, but with an action to research what the property is and why it was added. 

History on the `proto` property:
* [r16038](https://core.trac.wordpress.org/changeset/16038) added this property.
* [r19501](https://core.trac.wordpress.org/changeset/19501) moved it from the `initialize()` method to `__get()`.
* When 3.1 released, the protocol handling switched to use `get_admin_url()`, `admin_url()`, etc functions instead of the `proto` property. But the property remained in the code, unused.

Would removing it be a BC break? If not used, then no. But if some plugin or theme in the wild uses it, then yes it is a BC break.

Let's remove it with this reasoning:
* It's unused in Core and none found in wporg search
* It was introduced in the 3.1 cycle but then replaced with URL get functions during that same cycle.
* It's reasonable then to say that it should have been removed in 3.1, but was missed. It appears to be dead code.

Reply:
I've just submitted a [PR](https://github.com/WordPress/wordpress-develop/pull/4133) that implements the changes proposed in the @hellofromtonya 's comment above. Thanks!
https://github.com/WordPress/wordpress-develop/pull/4133